### PR TITLE
Emitter Rejiggering

### PR DESCRIPTION
**Pull request by Geevies**
* Emitters have received new sprites.
* Emitters can now be locked even when not firing. This prevents some forms of manipulation.
* Emitters can now have their barrels removed and replaced.
* Emitter barrels can now be placed on a plasteel stock, wired up, powered up, and used as a makeshift laser rifle.
* Emitters no longer need to be welded to the floor, simply bolting them is enough.

image
image
image
image

### CODE REVIEW

**Comment by Geevies** [comment]
!review

**Comment by MarinaGryphon** [comment]
Would kind of like to see this integrated with the modular guns system.

**Comment by alsoandanswer** [comment]
These sprites don't look so good. For some constructive criticism, it's button shaded and has gradients, both of which make it look like a codersprite. I preferred the old ones.

**Comment by Geevies** [comment]
> 
> 
> Would kind of like to see this integrated with the modular guns system.

Unfortunately my motivation to do this is as empty as my paypal account.

> 
> 
> These sprites don't look so good. For some constructive criticism, it's button shaded and has gradients, both of which make it look like a codersprite. I preferred the old ones.

Could you provide better looking sprites?

**Comment by alsoandanswer** [comment]
Yeah, sure.

File: code/modules/power/singularity/emitter.dm [hunk 1]
```diff
@@ -175,81 +198,94 @@
...
 		return
+
+	if(W.isscrewdriver())
+		if(!barrel)
+			to_chat(user, SPAN_WARNING("\The [src] doesn't have a barrel!"))
+			return
+		barrel_secured = !barrel_secured
+		var/others_msg = "\The [user] [barrel_secured ? "" : "un"]secures \the [src]'s barrel."
+		var/self_msg = "You [barrel_secured ? "" : "un"]secure \the [src]'s barrel."
+		user.visible_message(SPAN_WARNING(others_msg), SPAN_WARNING(self_msg), range = 3)
```
Comment (NonQueueingMatt) [inline code/modules/power/singularity/emitter.dm:257]: would appreciate it if this followed this convention: bold the name with no spans for others, leave as is for self

File: code/modules/power/singularity/emitter.dm [hunk 2]
```diff
@@ -175,81 +198,94 @@
...
+			electrocute_mob(user, powernet, src)
+			return
+		user.visible_message(SPAN_WARNING("\The [user] starts prying out \the [src]'s barrel..."), SPAN_NOTICE("You start prying out \the [src]'s barrel..."), range = 3)
+		if(do_after(user, 50, TRUE, src))
+			if(!barrel)
+				to_chat(user, SPAN_WARNING("\The [src] doesn't have a barrel!"))
+				return
+			if(barrel_secured)
+				to_chat(user, SPAN_WARNING("\The [src] barrel is still secured within its housing!"))
+				return
```
Comment (NonQueueingMatt) [inline code/modules/power/singularity/emitter.dm:279]: recheck the active condition here

File: code/modules/power/singularity/emitter.dm [hunk 3]
```diff
@@ -269,4 +305,148 @@
...
+						return
+					user.visible_message(SPAN_WARNING("\The [user] adds \the [ES] to \the [src]."), SPAN_NOTICE("You add \the [ES] to \the [src]."), range = 3)
+					icon_state = "emitter_barrel-1"
+					build_state = BARREL_STOCKED
+					qdel(ES)
+		if(BARREL_STOCKED)
+			if(W.iscoil())
+				var/obj/item/stack/cable_coil/CC = W
+				if(CC.amount < 5)
+					to_chat(user, SPAN_WARNING("You need atleast 5 lengths of coil in \the [CC]!"))
```
Comment (NonQueueingMatt) [inline code/modules/power/singularity/emitter.dm:338]:
```suggestion
					to_chat(user, SPAN_WARNING("You need at least 5 lengths of coil in \the [CC]!"))
```

File: code/modules/power/singularity/emitter.dm [hunk 4]
```diff
@@ -269,4 +305,148 @@
...
+					return
+				user.visible_message(SPAN_WARNING("\The [user] starts wiring up \the [src]..."), SPAN_NOTICE("You start wiring up \the [src]..."), range = 3)
+				if(do_after(user, 50, TRUE, src))
+					if(build_state != BARREL_STOCKED)
+						return
+					if(CC.amount < 5)
+						to_chat(user, SPAN_WARNING("You need atleast 5 lengths of coil in \the [CC]!"))
```
Comment (NonQueueingMatt) [inline code/modules/power/singularity/emitter.dm:345]:
```suggestion
						to_chat(user, SPAN_WARNING("You need at least 5 lengths of coil in \the [CC]!"))
```

File: code/modules/power/singularity/emitter.dm [hunk 5]
```diff
@@ -269,4 +305,148 @@
...
+						build_state = BARREL_UNMODIFIED
+						icon_state = "emitter_barrel"
+		if(BARREL_WIRED)
+			if(W.isscrewdriver())
+				user.visible_message(SPAN_WARNING("\The [user] starts fastening and securing all \the [src]'s attachments..."), SPAN_NOTICE("You start fastening and securing all \the [src]'s attachments..."), range = 3)
+				playsound(get_turf(src), W.usesound, 75, TRUE)
+				if(do_after(user, 50, TRUE, src))
+					if(build_state != BARREL_WIRED)
+						return
+					user.visible_message(SPAN_WARNING("\The [user] fastens and secures all \the [src]'s attachments."), SPAN_NOTICE("You fasten and secure all \the [src]'s attachments."), range = 3)
```
Comment (NonQueueingMatt) [inline code/modules/power/singularity/emitter.dm:373]:
```suggestion
					user.visible_message(SPAN_WARNING("\The [user] fastens and secures all \the [src]'s attachments."), SPAN_NOTICE("You fasten and secure all of \the [src]'s attachments."), range = 3)
```

File: code/modules/power/singularity/emitter.dm [hunk 6]
```diff
@@ -269,4 +305,148 @@
...
+						user.visible_message(SPAN_WARNING("\The [user] removes the stock from \the [src]."), SPAN_NOTICE("You remove the stock from \the [src]."), range = 3)
+						var/obj/item/emitter_stock/ES = new /obj/item/emitter_stock(get_turf(src))
+						ES.build_state = STOCK_BOTTOMED
+						ES.icon_state = "emitter_stock-1"
+						user.put_in_hands(ES)
+						build_state = BARREL_UNMODIFIED
+						icon_state = "emitter_barrel"
+		if(BARREL_WIRED)
+			if(W.isscrewdriver())
+				user.visible_message(SPAN_WARNING("\The [user] starts fastening and securing all \the [src]'s attachments..."), SPAN_NOTICE("You start fastening and securing all \the [src]'s attachments..."), range = 3)
```
Comment (NonQueueingMatt) [inline code/modules/power/singularity/emitter.dm:368]:
```suggestion
				user.visible_message(SPAN_WARNING("\The [user] starts fastening and securing all of \the [src]'s attachments..."), SPAN_NOTICE("You start fastening and securing all \the [src]'s attachments..."), range = 3)
```

File: code/modules/power/singularity/emitter.dm (hunk 1 above)
Comment (NonQueueingMatt) [reply code/modules/power/singularity/emitter.dm:257]: this applies to like every visible message here

**Comment by skull132** [comment]
I am absolutely not a fan of these sprites.

**Comment by NonQueueingMatt** [comment]
> 
> 
> I am absolutely not a fan of these sprites.

same

**Comment by skull132** [comment]
Stale since July. Remove the sprites and update and we can easily merge this one.

**Pull request closed without merging**